src/parser/core/modules/ActionWindow/evaluators/ExpectedGcdCountEvaluator.tsx

```diff
@@ -6,9 +6,23 @@ import {EvaluatedAction} from '../EvaluatedAction'
 import {HistoryEntry} from '../History'
 import {EvaluationOutput, WindowEvaluator} from './WindowEvaluator'
 
+// Extremely conservative weave delay to prevent
+// any possibility of undercounting expected GCDs
+const weaveDelay = 250
+
 // exported for use in AllowedGcdsOnlyEvaluator
-export function calculateExpectedGcdsForTime(defaultExpected: number, gcdEstimate: number, start: number, end?: number) {
-	return Math.min(defaultExpected, Math.ceil(((end ?? start) - start) / gcdEstimate))
+export function calculateExpectedGcdsForTime(defaultExpected: number, gcdEstimate: number, hasStacks: boolean, start: number, end?: number) {
+	let usableWindow = (end ?? start) - start
+
+	// Buffs with stacks have durations tightly coupled to the GCD
+	// and do not benefit from accounting for weave delay
+	if (!hasStacks) {
+		usableWindow -= weaveDelay
+	}
+
+	usableWindow = Math.max(usableWindow, 1)
+
+	return Math.min(defaultExpected, Math.ceil(usableWindow / gcdEstimate))
```

Comment:
I would prefer to see the window adjustment logic in the `calculateExpectedGcdsForWindow` function instead, and this function should just naively take start and end values of a time range, and not care about whether the underlying buff has stacks.  As is, this change is going to cause a change to the way Dragon Sight calculates "forgiven" GCDs when the tether partner dies and may result in forgiving one GCD too few (or having a really counterintuitive need to pass "true" hasStacks on that forgiveness calculation).